scraper/cazy_webscraper.py

```diff
@@ -539,13 +517,42 @@ def parse_family_pages(family_url, family_name, cazy_home, logger):
                 f"No protein records for CAZy family {family_name} will be retried."
             )
         )
-        return []
+        return family
+
+    protein_page_urls = get_protein_page_urls(
+        first_pagination_url,
+        first_pagination_page[0],
+        cazy_home
+    )
+
+    for protein in tqdm(
+        (
+            y for x in (
+                parse_proteins(url, family_name, logger) for url in protein_page_urls
+            ) for y in x
+        ),
+        total=len(protein_page_urls),
+        desc=f"Scraping protein pages for {family_name}",
+    ):
+        family.members.add(protein)
+
+    return family
 
+
+def get_protein_page_urls(first_pagination_url, first_pagination_page, cazy_home):
```

Comment:
I think I like this approach but you could, conceivably, have used a generator function (more memory efficient) instead and had a separate function that returned the expected number of pages for `tqdm`. This approach should also work, and probably won't be a bottleneck, so no need to optimise yet ;)